File: core/src/main/java/com/pluxity/facility/line/dto/LineResponse.java
package com.pluxity.facility.line.dto;

import com.fasterxml.jackson.annotation.JsonUnwrapped;
import com.pluxity.facility.Facility;
import com.pluxity.facility.line.Line;
import com.pluxity.global.response.BaseResponse;
import java.util.List;
import java.util.stream.Collectors;

public record LineResponse(
        Long id,
        String color,
        String name,
        List<Long> stationIds,
        @JsonUnwrapped BaseResponse baseResponse) {
    public static LineResponse from(Line line) {
        return new LineResponse(
                line.getId(),
                line.getColor(),
                line.getName(),
                line.getStations().stream().map(Facility::getId).collect(Collectors.toList()),
                BaseResponse.of(line));
    }
}

File: core/src/main/kotlin/com/pluxity/station/dto/LineResponse.kt
package com.pluxity.station.dto

import com.fasterxml.jackson.annotation.JsonUnwrapped
import com.pluxity.global.response.BaseResponse
import com.pluxity.global.response.toBaseResponse
import com.pluxity.station.Line

data class LineResponse(
    val id: Long?,
    val color: String?,
    val name: String,
    val stationIds: List<Long>,
    @field:JsonUnwrapped val baseResponse: BaseResponse,
)

fun Line.toLineResponse(): LineResponse =
    LineResponse(
        this.id,
        this.color,
        this.name,
        this.getStations().mapNotNull { it.id },
        this.toBaseResponse(),
    )
